test/integration/specs/kbaseCellToolbarMenu_test.js

```diff
@@ -0,0 +1,327 @@
+                merge(obj1Value, obj2Value, keyStack);
+                keyStack.pop();
+            } else {
+                console.error('UNMERGABLE', obj1Type, obj1Value);
+                throw new Error('Unmergable at ' + keyStack.join('.') + ':' + key);
+            }
+        });
+    }
+
+    const base = JSON.parse(JSON.stringify(listOfObjects[0]));
+    for (let i = 1; i < listOfObjects.length; i += 1) {
+        merge(base, listOfObjects[i], []);
+    }
+    return base;
+}
+
+async function waitForCell(notebookContainer, cellIndex){
+    return await browser.waitUntil(async () => {
+        const cell = await notebookContainer.$(`.cell:nth-child(${cellIndex})`);
+        return cell;
+    });
+}
+
+async function waitForCellWithTitle(container, cellIndex, title) {
+    const cell = await waitForCell(container, cellIndex);
+    await browser.waitUntil(async () => {
+        const titleElement = await cell.$('[data-element="title"]');
+        const text = await titleElement.getText();
+        return text === title;
+    });
+    return cell;
+}
+
+async function waitForCellWithBody(container, cellIndex, bodyText) {
+    const cell = await waitForCell(container, cellIndex);
+    await browser.waitUntil(async () => {
+        const element = await cell.$('.text_cell_render.rendered_html');
+        const text = await element.getText();
+        return text === bodyText;
+    });
+    return cell;
+}
+
+async function selectCell(container, cellIndex, title) {
+    const cell = await waitForCellWithTitle(container, cellIndex, title);
+
+    // Make sure not selected.
+    // We do this by inspecting the border.
+    await browser.waitUntil(async () => {
+        const borderStyle = await cell.getCSSProperty('border');
+        return borderStyle.value === testData.common.unselectedBorder;
+    });
+
+    // Click on title area to select cell.
+    const titleArea = await cell.$('.title-container');
+    await clickWhenReady(titleArea);
+
+    await browser.waitUntil(async () => {
+        const borderStyle = await cell.getCSSProperty('border');
+        return borderStyle.value === testData.common.selectedBorder;
+    });
+
+    return cell;
+}
+
+async function selectCellWithBody(container, cellIndex, body) {
+    const cell = await waitForCellWithBody(container, cellIndex, body);
+
+    // Make sure not selected.
+    // We do this by inspecting the border.
+    await browser.waitUntil(async () => {
+        const borderStyle = await cell.getCSSProperty('border');
+        return borderStyle.value === testData.common.unselectedBorder;
+    });
+
+
+    // Click on title area to select cell.
+    const bodyArea = await cell.$('.text_cell_render.rendered_html');
+    await clickWhenReady(bodyArea);
+
+    await browser.waitUntil(async () => {
+        const borderStyle = await cell.getCSSProperty('border');
+        return borderStyle.value === testData.common.selectedBorder;
+    });
+
+    return cell;
+}
```

Comment:
This is almost identical to the function above. I would refactor this code to create a more general `selectCellPart` function, which has an additional param that indicates whether the title or the body should be selected.

Reply:
There will be an update to this. Not quite the way you suggested, but close in spirit.